src/chainstate/stacks/db/accounts.rs

```diff
@@ -388,6 +388,59 @@ impl StacksChainState {
             .expect("FATAL: failed to set account nonce")
     }
 
+    /// Increase a STX lock up for PoX.  Does NOT touch the account nonce.
+    /// Returns Ok( account snapshot ) when successful
+    ///
+    /// # Errors
+    /// - Returns Error::PoxExtendNotLocked if this function was called on an account
+    ///     which isn't locked. This *should* have been checked by the PoX v2 contract,
+    ///     so this should surface in a panic.
+    pub fn pox_lock_increase_v2(
+        db: &mut ClarityDatabase,
+        principal: &PrincipalData,
+        new_total_locked: u128,
+    ) -> Result<STXBalance, Error> {
+        assert!(new_total_locked > 0);
+
+        let mut snapshot = db.get_stx_balance_snapshot(principal);
+
+        if !snapshot.has_locked_tokens() {
```

Comment:
I find myself wondering whether or not these `Err(..)` paths are ever reachable.  If the Clarity code for `pox-2` is correct, then these should never be hit, right?

Reply:
Ah, I see we `panic()` later in the special case handler.